### DISCUSSION TITLE
Function to set a window's monitor

### DISCUSSION THREAD

**Discussion by SolarLune** [hajimehoshi/ebiten#1834] [Ideas]
Hello~

I was wondering if it would be possible to add a function to set the monitor a window appears on? That would be helpful for me personally, as I have two monitors and running an Ebiten app always places it directly on my IDE.

**Comment by hajimehoshi** [comment]
Sure, let me think :-) https://github.com/hajimehoshi/ebiten/issues/1835

**Reply by hajimehoshi** [reply]
Even with the current Ebiten, you can specify the position by `SetWindowPosition`. Would this be enough?

**Reply by SolarLune** [reply]
Unfortunately, no - 0,0 corresponds to the current monitor's top-left corner, so I can't move a window to the left of that (my secondary monitor is to the left of my primary one). 

Also, since 0,0 is the monitor's top-left corner, you can't really know which monitor a window currently is on.

**Reply by hajimehoshi** [reply]
The default position is now not (0, 0) (as of 2.2.0-rc.1). Have you tried?

> Also, since 0,0 is the monitor's top-left corner, you can't really know which monitor a window currently is on.

If you know the sizes of the monitor, you can specify the window position, right?

**Reply by SolarLune** [reply]
I updated to v2.3.0-alpha and 0,0 is still the top-left corner of whichever monitor the window's on. And yeah, knowing the size of the monitors would allow me to work with it.

**Reply by hajimehoshi** [reply]
Ah yes, (0, 0) is still the top-left corner on Linux. The initial monitor should be the current active monitor but probably this is what you have already indicated: "an Ebiten app always places it directly on my IDE".

> And yeah, knowing the size of the monitors would allow me to work with it.

Sure, let me think.

**Comment by airtonix** [comment]
Related to this is the opposite:

> how to know which monitor the window is on? 

since `ebitten.GetWindowPosition()` on linux doesn't return the overall accumulative position it's impossible to answer this question.

if screen one is `1024x768` and screen two is `1920x1080` and as a human i can see the window is on screen two at `300x240`, then i expect: 

```go
x,y := ebitten.GetWindowPosition() 
fmt.Println(x,y)
# prints 1324, 240
```

but instead it prints

```go
x,y := ebitten.GetWindowPosition() 
fmt.Println(x,y)
# prints 300, 240
```

**Reply by airtonix** [reply]
turns out it does, but it doesn't respect the arrangement defined (by xrandr i think).

**Reply by airtonix** [reply]
see https://github.com/hajimehoshi/ebiten/issues/2876

**Reply by airtonix** [reply]
and https://github.com/hajimehoshi/ebiten/issues/2877
